typescript/sdk/src/ism/HyperlaneIsmFactory.ts

```diff
@@ -54,20 +46,42 @@ import {
   DeployedIsmType,
   IsmConfig,
   IsmType,
-  ModuleType,
   MultisigIsmConfig,
-  OpStackIsmConfig,
   RoutingIsmConfig,
   RoutingIsmDelta,
-  ismTypeToModuleType,
 } from './types';
+import { routingModuleDelta } from './utils';
 
 export class HyperlaneIsmFactory extends HyperlaneApp<ProxyFactoryFactories> {
   // The shape of this object is `ChainMap<Address | ChainMap<Address>`,
   // although `any` is use here because that type breaks a lot of signatures.
   // TODO: fix this in the next refactoring
   public deployedIsms: ChainMap<any> = {};
 
+  // upon initialization, HyperlaneDeployer will bind itself to HyperlaneIsmFactory
```

Comment:
could we alternatively have HyperlaneIsmFactory extend HyperlaneDeployer?

Reply:
probably could, would need to create a new hyperlaneapp as a field inside this factory class and update refs but doable

decided to go with binding because I wasn't sure of the ramifications of changing this class from an app -> deployer